lib/peg.d.ts

```diff
@@ -51,6 +51,23 @@ declare namespace ast {
     location: LocationRange;
   }
 
+  /**
+   * Type of the classes field on a Grammar node. Not quite the same as
+   * CharacterClass (`parts` was renamed to `value`).
+   */
+  interface CharacterRange {
+    value: (string[] | string)[];
+    inverted: boolean;
+    ignoreCase: boolean;
+  }
+
+  type Expectation =
+    | { type: "any" }
+    | { type: "literal"; value: string; ignoreCase: boolean }
+    | { type: "rule"; value: string }
+    | CharacterRange & { type: "class" }
+    ;
+
```

Comment:
@hildjj 

I don't like this change (too much confusion, and overlap between `ast.Expectation`, and `parser.Expectation`) but I think its a correct description of what's going on.

Previously, `Grammar.Expectations` was typed as `parser.Expectation[]`, but that's not right, because `parser.ClassExpectation`, has a `value` field, rather than a `parts` field.

Also, `parser.Expectation` supports types `any`, `class`, `end`, `literal`, and `other`, while `Grammar.expectations[]` supports types `any`, `class`, `literal` and `rule` - with assertions for any other types that appear.

Similarly `Grammar.classes[]` has `value`, rather than `parts`.

This seems pointlessly confusing. Essentially generate-bytecode.js takes an object with a `parts` array, and converts it to an object with a `value` array ([here](https://github.com/peggyjs/peggy/blob/main/lib/compiler/passes/generate-bytecode.js#L287-L291) and [here](https://github.com/peggyjs/peggy/blob/main/lib/compiler/passes/generate-bytecode.js#L1016-L1020). And then generate-js.js [converts it back again](https://github.com/peggyjs/peggy/blob/main/lib/compiler/passes/generate-js.js#L1009).

It would be nice to fix it so that `Grammar.classes` and `Grammar.expectations` both use `parts` rather than `values` (and maybe rename `rule` as `other`) - but presumably that could break plugins?

If we're stuck with this, I think I should rename these to something like `GrammarCharacterClass` and `GrammarExpectation` so its clear they're just for this one use...

Reply:
Let's go ahead and at least do the rename.  It will be easier to see the consolidation change after.